lib/fluent/plugin/in_systemd.rb

```diff
@@ -29,6 +29,7 @@ def start
       @running = true
       pos_writer.start
       @thread = Thread.new(&method(:run))
+      @thread.abort_on_exception = true
```

Comment:
Hah seems we both misread that doc. To highlight: **When set to true, causes** _all threads to abort_ **if an exception is raised in thr** - i.e. the thread that you're setting this on, not all of them.